include/tudat/simulation/environment_setup/createSystemModel.h

```diff
@@ -195,8 +224,10 @@ class BodyPanelSettings
 public:
     BodyPanelSettings( const std::shared_ptr< BodyPanelGeometrySettings > panelGeometry,
                        const std::shared_ptr< BodyPanelReflectionLawSettings > reflectionLawSettings,
-                       const std::string panelTypeId = "" ):
-        panelGeometry_( panelGeometry ), reflectionLawSettings_( reflectionLawSettings ), panelTypeId_( panelTypeId )
+                       const std::string panelTypeId = "",
+                       const std::shared_ptr< MaterialProperties > materialProperties = nullptr ):
```

Comment:
This is a great modification! The interfaces were becoming a mess, and thi is a nice solution